hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -1589,8 +1582,13 @@ protected Option<HoodieBaseFile> fetchLatestBaseFile(String partitionPath, Strin
    * @return File Slice if present
    */
   public Option<FileSlice> fetchLatestFileSlice(String partitionPath, String fileId) {
-    return Option
-        .fromJavaOptional(fetchLatestFileSlices(partitionPath).filter(fs -> fs.getFileId().equals(fileId)).findFirst());
+    try {
+      readLock.lock();
+      return Option
+          .fromJavaOptional(fetchLatestFileSlices(partitionPath).filter(fs -> fs.getFileId().equals(fileId)).findFirst());
```

Comment:
Isn't the read lock already acquired in line 1588?

Reply:
No